### ISSUE TITLE
Set $ErrorActionPreference to Stop in PowerShell function templates code

### ISSUE THREAD

**Issue by AnatoliB** [Azure/azure-functions-powershell-worker#364]
**Problem**

By PowerShell design, the default value of `$ErrorActionPreference` is `'Continue'`. New PowerShell users who are not familiar with the concept of non-terminating errors often get confused by the result: the PowerShell function execution seems to be completed successfully but has not in fact performed what it was supposed to perform, and the only way to notice this is to look at the logs.

This problem occasionally affects even experienced PowerShell users, and it seems to be a frequent annoyance for beginners, so it is worth mitigating. At the same time, we don't want to surprise experienced PowerShell users with by modifying the default behavior or introducing any other breaking changes, and we want to avoid creating Functions-specific abstractions on top of PowerShell.

**Proposal**

When a new PowerShell function is added from a template, the auto-generated code should include the following:

``` PowerShell
# By default, stop on any error
# (for more details, see <link to the doc, e.g. https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_preference_variables#erroractionpreference>)
$ErrorActionPreference = 'Stop'
```

As a result, this will make users aware of the difference between terminating vs. non-terminating errors, invite them to learn more if interested, and set a more intuitively expected behavior. At the same time, users familiar with PowerShell who insist on any different behavior will have an opportunity to review, change, or delete this code.

**Comment by JustinGrote** [comment]
Makes sense to me and wouldn't be a breaking change unless someone has some edge case where they are doing code generation off the existing templates and expecting a certain behavior (pretty unlikely)

**Comment by anirudhgarg** [comment]
@AnatoliB  are we planning to add this to the profile.ps1 ?

**Comment by AnatoliB** [comment]
@anirudhgarg I was actually thinking about `run.ps1` instead. Technically, both `profile.ps1` and `run.ps1` would work. However, the decision to use a non-default `$ErrorActionPreference` should normally be coupled with the content of a specific script, i. e. you would want to author your script differently depending on the `$ErrorActionPreference` value. Also, `profile.ps1` is more implicit and less visible. I doubt many PowerShell users set `$ErrorActionPreference` in their `profile.ps1` (probably for the same reasons), so I would just follow the existing practice.

**Comment by JustinGrote** [comment]
My vote is for `profile.ps1` by default in a new function. Most functions are so short lived you want them to "fail fast" and not generate a bunch of non-terminating exceptions

**Comment by rgommers** [comment]
This is still a problem that causes a lot of lost time in 2023.

**Comment by AnatoliB** [comment]
@Francisco-Gamino @andystaples 
One reason why this has not been done yet is that this would be a breaking change. Indeed, if we follow the original proposal, this change rollout will not be tied to the language version, so it may break existing automated workflows that create a new function from templates. So, what if we make it happen for PS 7.4+ only, like this:

``` PowerShell
# By default, stop on any error
# (for more details, see <link to the doc, e.g. https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_preference_variables#erroractionpreference>)
if ($PSVersionTable.PSVersion -ge [version]'7.4') {
    $ErrorActionPreference = 'Stop'
}
```

When PS 7.2 is out of support, we can remove the condition. What do you think? Adding the triage tag to resurrect this discussion.

**Comment by Francisco-Gamino** [comment]
This needs to be validated with different version of PowerShell.

**Comment by Francisco-Gamino** [comment]
@andystaples -- These are the files that need to be updated:

Host (this is for the Portal scenario)
[azure-functions-host/src/WebJobs.Script/FileProvisioning/PowerShell/profile.ps1 at dev · Azure/azure-functions-host (github.com)](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/FileProvisioning/PowerShell/profile.ps1)

VSCode Extension 
[vscode-azurefunctions/src/commands/createNewProject/ProjectCreateStep/PowerShellProjectCreateStep.ts at 7d04a34685a8c6f53863f83b85088ca624c52b1c · microsoft/vscode-azurefunctions (github.com)](https://github.com/microsoft/vscode-azurefunctions/blob/7d04a34685a8c6f53863f83b85088ca624c52b1c/src/commands/createNewProject/ProjectCreateStep/PowerShellProjectCreateStep.ts#L21-L43)

Core Tools 
[azure-functions-core-tools/src/Azure.Functions.Cli/StaticResources/profile.ps1 at dev · Azure/azure-functions-core-tools (github.com)](https://github.com/Azure/azure-functions-core-tools/blob/dev/src/Azure.Functions.Cli/StaticResources/profile.ps1)
